samples/MyLibrary/Class1.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Hosting.Internal;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Localization;
+
+namespace MyLibrary
+{
+    public class Class1
+    {
+        private IStringLocalizer<Class1> _localizer;
+
+        public Class1()
+        {
+            var services = new ServiceCollection();
```

Comment:
I'm not sure that's possible to do in general due to ResourcePath. If the ResourcePath being used by your library isn't the same as the one being used by your main app how are you going to know where to look for it? That information is normally kept in the `ResourceManagerStringLocalizerFactory`, but we can't rely on it being the same between assemblies. That may mean that this whole exercise is in vain.